lib/pdf_abr_writer.rb

```diff
@@ -40,42 +43,23 @@ def generate_pdf(force_write = false, for_printer = false)
         form.save_as(pdf_file_path, flatten: false)
       end
 
-      
-      unless deliver_to_elections_office_via_email?
-        # Temp solution for making sure every PDF includes address
-        html_string = "Sign and return this form to:<br/><br/><p style='font-size: 24px; line-height: 1.6em;'>#{delivery_address}</p>"
-        #0. generate address pdf
-        pdf = WickedPdf.new.pdf_from_string(
-          html_string,
-          :disable_internal_links         => false,
-          :disable_external_links         => false,
-          :encoding => 'utf8',
-          :locale=>locale,
-          :page_size => locale.to_s == "en" ? "Letter" : "A4"
-        )
-
-        File.open(pdf_delivery_address_path, "w") do |f|
-          f << pdf.force_encoding('UTF-8')
-        end
-      end
 
       # 1. generate xfdf
       xfdf_contents = "<?xml version=\"1.0\"?><xfdf xmlns=\"http://ns.adobe.com/xfdf/\"><fields>"
       pdf_values.each do |k, v|
         xfdf_contents += "\n<field name=\"#{k}\"><value>#{v}</value></field>"
       end
+      pdf_cover_values.each do |k, v|
+        xfdf_contents += "\n<field name=\"#{k}\"><value>#{v}</value></field>"
+      end
       xfdf_contents += "</fields></xfdf>"
       
       File.open(pdf_xfdf_path, "w+") do |f|
         f.write xfdf_contents
       end
-      `pdftk #{(voter_signature && !voter_signature.voter_signature_image.blank? ? pdf_file_path : pdf_template_path).to_s} fill_form #{pdf_xfdf_path} output #{pdf_file_path}-tmp flatten`        
-      if deliver_to_elections_office_via_email?
-        # no cover page needed
-        `cp #{pdf_file_path}-tmp #{pdf_file_path}`
-      else
-        `pdftk #{pdf_delivery_address_path} #{pdf_file_path}-tmp output #{pdf_file_path}`
-      end
+      `pdftk #{(voter_signature && !voter_signature.voter_signature_image.blank? ? pdf_file_path : pdf_template_with_cover_path).to_s} fill_form #{pdf_xfdf_path} output #{pdf_file_path}-tmp flatten`        
+
+      `cp #{pdf_file_path}-tmp #{pdf_file_path}`  
```

Comment:
Thinking out loud a bit on this. We want to reduce # files and file ops, so the two cases are
1. Signed file
1a. Generate file w/sig (pdf_template_path) => signed file
1b. Fill signed file with pdftk (from signed file template)
2. Unsigned
2a. Fill unsigned file with pdftk (pdf_template_with_cover_path).

So I think we can change this to:

1a. from pdf_template, create "#{pdf_file_path}-signed"
1b. from "#{pdf_file_path}-signed" fill_form to create "#{pdf_file_path}
Or 
2 from pdf_template_with_cover_path fill_form to create "#{pdf_file_path}
And we can remove the cp statement?
It'll require changing the set of file names that get cleaned up at the end.

Reply:
Yah.  I saw you had a comment in to that effect somewhere and I *had* had it optimized for that, but then it was broken when I tried to overwrite an input file as output for the email condition.  

I will re-minimize the file ops.

Reply:
Have done